### ISSUE TITLE
match history and current ratings do not align

### ISSUE THREAD

**Issue by ghost** [jrmitch120/ChallengeBoard#13]
Something is not right here.
Rating doesn't align with the last match. Thoughts on how was this possible?
Any chance there was a recent deployment with some new stuff that might cause that?

Fortuantely it seems that the current rating is correct and so are -/+ deltas in the match history.
Though ratings in the match history are totally messed up.

Following screenshot was taken from http://www.challengeboards.net/boards/competitors/2179/profile/8575

![image](https://cloud.githubusercontent.com/assets/9743319/6748936/9ff8c88a-ceeb-11e4-9b99-a544e778191a.png)

**Comment by jrmitch120** [comment]
On a cursory look, this is displaying only official (resolved) matches.

So there is/was likely some "unresolved" matches that are "invisible" until they resolve themselves.

So in plainer terms, it looks like he played a serious of matches: A, B, C, D, E, F.

A, B, E, & F are validated by his opponent, so they become official and are displayed.  C, D, and not validated by the opponent, so they remain hidden until they timeout and become official.

As the matches become official, this should math-out correctly.  I'm open on whether we want to display the unofficial matches here.  It might make if clearer.  Does that makes sense?

**Comment by ghost** [comment]
Yeah, I think your explanation is correct. What I don't understand though is the fact, that many games verified by the opponent stay in the backlog (rightfully so, cause they are blocked by other unverified games). Only some of them have the power to jump from "pending matches" to "recent matches" (and this "power" is essentially the part I don't get). What is the criteria for this? How can E and F (verified) leave "pending matches" with C and D (not verified) still blocking them?

**Comment by jrmitch120** [comment]
Once validated (accepted, rejected, withdrawn, or auto-accepted via timeout) a match has it's `resolved` property set.  This indicates that E and F are no longer pending.  This can't be undone, so they appear in the official listing of matches.  

However, E and F haven't been applied to the user's official rating yet.  This will happen once everything in front of them clears out.  If a match in front of E and F is rejected, their ratings will be recalculated accordingly.  

Only once everything in front of them has been resolved, E and F be applied to the user's official rating.

**Comment by ghost** [comment]
Gotcha. But wouldn't this be more clear if `resolved` matches appeared in the official listings only after everything before them clears out? Don't you think that situation on the screenshot is very confusing?

**Comment by jrmitch120** [comment]
Yes :)  We'll make it so!
